File: src/easy/palindrome-number/1.ts
/*
 * @lc app=leetcode id=9 lang=typescript
 *
 * [9] Palindrome Number
 */

// @lc code=start

function isPalindrome(x: number): boolean {
  if (x < 0) {
    return false;
  }

  let original = x;
  let reverse = 0;
  while (x > 0) {
    reverse += (x % 10) * Math.pow(10, Math.floor(Math.log10(x)));
    x = Math.floor(x / 10);
  }
  return reverse == original;
}
// @lc code=end

File: src/easy/palindrome-number/1.go
package palindromenumber

import "strconv"

/*
 * @lc app=leetcode id=9 lang=golang
 *
 * [9] Palindrome Number
 * two pionter approach
 */

// @lc code=start
func isPalindrome(x int) bool {
	str := strconv.Itoa(x)
	l := 0
	r := len(str) - 1
	for r > l {
		if str[l] != str[r] {
			return false
		}
		l++
		r--
	}
	return true
}

// @lc code=end
